src/WebJobs.Script.WebHost/WebHostServiceCollectionExtensions.cs

```diff
@@ -227,6 +227,7 @@ public static void AddWebJobsScriptHost(this IServiceCollection services, IConfi
             services.ConfigureOptions<AzureMonitorLoggingOptionsSetup>();
             services.AddHostingConfigOptions(configuration);
             services.ConfigureOptions<ExtensionRequirementOptionsSetup>();
+            services.Configure<HostMetricsProviderOptions>(configuration.GetSection("HostMetricsProvider")); // todo: use options setup?
```

Comment:
Yes it might make sense to use an options setup, similar to FlexConsumptionMetricsPublisherOptionsSetup, to centralize the initialization logic.

What external entity will be configuring this value though? Are you doing this in case we need to configure via hosting config across the fleet? I don't think we need individual apps to configure this separately.

Reply:
`.Configure` _is_ an options setup. This adds an `IConfigureOptions` under the hood which calls `IConfiguration.Bind(options)`.

The question would then be: any reason we don't want config binding here?

Reply:
Definitely not for individual app usage, but I wanted to give us an option to change the interval value across for the fleet in case 30 seconds was too long or not enough. I just don't love the idea of having a hard coded value for something like this. 

If hosting config makes more sense, I can move the interval there and scrap `HostMetricsProviderOptions`

Reply:
@mathewc @kshyju per the latest conversations on this (see: https://github.com/Azure/azure-functions-host/pull/11019?notification_referrer_id=NT_kwDOACGNebMxNTk5MTk2OTEwNjoyMTk4OTA1#issuecomment-2843191639). I feel this is no longer needed as we're using the flex publisher options again instead of introducing a new one. However, if you still feel this needs to be in host config let me know.